apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.Data/SpannerCommand.cs

```diff
@@ -235,10 +236,14 @@ public override int ExecuteNonQuery()
             }
 
             // Make the request.  This will commit immediately or not depending on whether a transaction was explicitly created.
-            await GetSpannerTransaction().ExecuteMutationsAsync(mutations, cancellationToken).ConfigureAwait(false);
-
-            // Return the number of records affected.
-            return mutations.Count;
+            var task = GetSpannerTransaction().ExecuteMutationsAsync(mutations, cancellationToken);
+            if (await Task.WhenAny(task, Task.Delay(CommandTimeout, cancellationToken)).ConfigureAwait(false) == task)
+            {
+                await task;
+                // Return the number of records affected.
+                return mutations.Count;
+            }
+            throw new SpannerException(ErrorCode.DeadlineExceeded, "The timeout of the SpannerCommand was exceeded.");
```

Comment:
System.TimeoutException? Not sure - probably worth checking what other database clients do here.